### ISSUE TITLE
DaMiRseq package

### ISSUE THREAD

**Issue by BioinfoMonzino** [Bioconductor/Contributions#272]
Update the following URL to point to the GitHub repository of
the package you wish to submit to _Bioconductor_

- Repository: https://github.com/BioinfoMonzino/DaMiRseq

Confirm the following by editing each check box to '[x]'

- [x] I understand that by submitting my package to _Bioconductor_,
  the package source and all review commentary are visible to the
  general public.

- [x] I have read the _Bioconductor_ [Package Submission][2]
  instructions. My package is consistent with the _Bioconductor_
  [Package Guidelines][1].

- [x] I understand that a minimum requirement for package acceptance 
  is to pass R CMD check and R CMD BiocCheck with no ERROR or WARNINGS. 
  Passing these checks does not result in automatic acceptance. The 
  package will then undergo a formal review and recommendations for 
  acceptance regarding other Bioconductor standards will be addressed.

- [x] My package addresses statistical or bioinformatic issues related
  to the analysis and comprehension of high throughput genomic data.

- [x] I am committed to the long-term maintenance of my package. This
  includes monitoring the [support site][3] for issues that users may
  have, subscribing to the [bioc-devel][4] mailing list to stay aware
  of developments in the _Bioconductor_ community, responding promptly
  to requests for updates from the Core team in response to changes in
  _R_ or underlying software.

I am familiar with the essential aspects of _Bioconductor_ software
management, including:

- [x] The 'devel' branch for new packages and features.
- [x] The stable 'release' branch, made available every six
      months, for bug fixes.
- [x] _Bioconductor_ version control using [Subversion][5]
  (optionally [via GitHub][6]).

For help with submitting your package, please subscribe and post questions
to the [bioc-devel][4] mailing list.

[1]: https://bioconductor.org/developers/package-guidelines/
[2]: https://bioconductor.org/developers/package-submission/
[3]: https://support.bioconductor.org
[4]: https://stat.ethz.ch/mailman/listinfo/bioc-devel
[5]: http://bioconductor.org/developers/how-to/source-control/
[6]: http://bioconductor.org/developers/how-to/git-mirrors/

**Comment by mtmorgan** [comment]
@LiNk-NY  will provide a review of your package. I note the following

- Please review the need for packages in the Depends: field -- these should only be packages the user needs to directly access to prepare input for or interpret output from your package. Likely most of these belong in the Imports: field, with appropriate additions to the NAMESPACE file.
- For interoperability and repeatability your package should expect input in the form of SummarizedExperiment.
- 'Pipeline' packages are seldom successful, because they necessarily make too many assumptions about the user data and stability of dependent packages. Start each function with rigorous tests on the pre-conditions / assumptions about the nature of the data (tests for 'data.frame', for instance, are not sufficient), and provide unit tests that ensure that these are implemented. Increase the robustness of your package by providing unit tests for each feature you are expecting to be available in dependent packages. 
- No need to start the Title: with "The DaMiRseq package - ", this will be obvious whenever the Title is exposed.
- Wrap the Description: and other fields (to 80 characters, indendting continuation lines by 4 spaces), so that it displays reasonbly

**Comment by mtmorgan** [comment]
I've taken over the review of this package from @LiNk-NY As a first step your package needs to pass R CMD build and check within the time frame indicated in the [Bioconductor guidelines](https://bioconductor.org/developers/package-guidelines/#correctness). It looks like your package took 44 minutes to build on windows, and timed out before working through the examples on during R CMD check. Please modify your vignette and examples to run in a timely fashion. Do so by creating a realistic but small example that illustrates your package functionality. Remember that examples and vignettes are required, and that 'don't run' chunks are not appropriate.

**Comment by BioinfoMonzino** [comment]
Thanks @LiNk-NY and @mtmorgan for your quick replies.
Regarding the suggestions of @LiNK_NY:
1.	We are modifying our functions, in order to accept a “SummarizedExperiment” objects as input.
2.	We are adapting the code and the DESCRIPTION file;
Regarding the requests of @mtmorgan:
We also observed that our package takes a long time to build on windows since examples and vignette are based on high-dimensional data. Anyway, we decided to use a real large dataset because the functions embedded in our package have been designed to handle such kind of data and cannot appropriately work with small dimensional data. However, we are going to replace the real data included in the package with a realistic but smaller “synthetic” dataset in order to solve the execution time issue. 
Thanks in advance,
Mattia and Luca

**Comment by BioinfoMonzino** [comment]
Dear @mtmorgan 
we accordingly modified: the DESCRIPTION and the NAMESPACE files;
 our functions in order to use  “SummarizedExperiment” objects;
 vignette and examples to reduce running time. 
We also created smaller but realistic examples that should properly display the package functionality and performed unit tests to ensure rigorous tests on functions.
Regards,
Luca and Mattia

**Comment by mtmorgan** [comment]
Please post questions related to the issue here, rather than on bioc devel. You said:

> Dear bioc-devel members,
> during the last build of our package (DaMiRseq package #272<https://github.com/Bioconductor/Contributions/issues/272>), the following error occurred:
> 
> Checking for bioc-devel mailing list subscription...
>     * ERROR: Maintainer must subscribe to the bioc-devel mailing list.
>       Subscribe here: https://stat.ethz.ch/mailman/listinfo/bioc-devel
> 
> although the maintainers have already subscribed to the "bioc-devel mailing list".
> We subscribed using the email address present in DESCRIPTION FILE ("Maintainer" field) and used for Bioiconductor accounts:
> Mattia Chiesa <mchiesa@ccfm.it>
> Luca Piacentini <luca.piacentini@ccfm.it>
> 
> However, we didn't receive any feedback.
> 
> Moreover, we got another error regarding FSelector package only during the build on Mac OS X Mavericks (10.9.5)/x86_64 system:
> 
> ERROR: dependency FSelector is not available for package DaMiRseq
> 
> * removing /private/var/folders/_l/1jpyvgt17qzgj1yz7f0z4np80000gr/T/RtmpEYhVBo/Rinst47b339db8791/DaMiRseq
> 
> Since the package is available in CRAN<https://cran.r-project.org/web/packages/FSelector/index.html> for this platform, we would like to know how to deal with this issue.
> 
> The whole build report is present in the following address:
> http://bioconductor.org/spb_reports/DaMiRseq_buildreport_20170221125412.html
> 
> Thanks in advance for any suggestion.
> Best Regards,
> Mattia Chiesa and Luca Piacentini.

But actually the bioc-devel mailing list rejected your post because mattia.chiesa@cardiologicomonzino.it is not a subscriber. I also tried to verify that the accounts you mention above exist, but they do not. The mailman application sends an acknowledgement, and you must reply to it; perhaps your mail system inappropriately labeled the request as 'spam'? Also, the maintainer field is supposed to be the SINGLE individual who takes ultimate responsibility for the package.

It seems that FSelector fails to build with the following error, also seen in the CRAN checks with gcc compiler.

```
** preparing package for lazy loading
Error : .onLoad failed in loadNamespace() for 'RWeka', details:
  call: .jnew("weka/core/WekaPackageClassLoaderManager")
  error: java.lang.UnsupportedClassVersionError: weka/core/WekaPackageClassLoaderManager : Unsupported major.minor version 51.0
ERROR: lazy loading failed for package ‘FSelector’
```

**Comment by BioinfoMonzino** [comment]
Dear @mtmorgan ,
we subscribed to [bioc-devel mailing](https://stat.ethz.ch/mailman/listinfo/bioc-devel) list with the following addresses:
Mattia Chiesa <mattia.chiesa@cardiologicomonzino.it>
Luca Piacentini <luca.piacentini@cardiologicomonzino.it>
We set these emails also in Bioconductor accounts and in the DESCRIPTION file.
![biocdevelsubscription](https://cloud.githubusercontent.com/assets/25501676/23214863/9f580304-f910-11e6-9865-ea9aaf21eb61.png)
We are waiting for the confirmation emails.  We also checked that the email was not inappropriately labelled as SPAM.
Regarding FSelector, we also checked in the CRAN repository. We found that the problem seems to be the version of Java installed in Mac OS X Mavericks (10.9.5)/x86_64 system, as reported in:
[https://www.r-project.org/nosvn/R.check/r-release-osx-x86_64-mavericks/RWeka-00install.html](https://www.r-project.org/nosvn/R.check/r-release-osx-x86_64-mavericks/RWeka-00install.html)

You can see also the following discussion:
[http://stackoverflow.com/questions/10382929/how-to-fix-java-lang-unsupportedclassversionerror-unsupported-major-minor-versi](http://stackoverflow.com/questions/10382929/how-to-fix-java-lang-unsupportedclassversionerror-unsupported-major-minor-versi)

How can we deal with this problem?
Thanks again.

**Comment by mtmorgan** [comment]
I can only confirm that the subscriptions are not present in the mailing list. Also please note the requirement that the maintainer contain a SINGLE email address; think of it as 'corresponding author'.

I'll discuss with other team members about the version of Java; ignore the error for now.

**Comment by BioinfoMonzino** [comment]
Dear @mtmorgan,
We are dealing with the tricky issue concerning the subscription to bioc-devel mailing list with our IT personnel, and it seems that there could be a problem with the @cardiologicomonzino.it and/or @r-project.org domain. However, we are going to bypass this concern by using a non-institutional email address.

**Comment by BioinfoMonzino** [comment]
Dear @mtmorgan ,
We got ‘TIMEOUT’ status during building on **Mac OS X Mavericks (10.9.5)/x86_64 system**, whereas Windows and Linux are ‘OK’. In particular, Windows takes about 6 minutes and Linux takes about 5 minutes for building; on the other hand, MAC OS seems to take 10 minutes and **0.15 seconds**, where the limit seems to be exactly 10 minutes:

 `* checking for file DaMiRseq/DESCRIPTION ... OK`
 `* preparing DaMiRseq:`
 `* checking DESCRIPTION meta-information ... OK`
 `* installing the package to build vignettes`
 `* creating vignettes ... ERROR`
 `TIMEOUT: R CMD build exceeded 10mins`

Should we just re-build? Otherwise, we must further modify our vignette to decrease the execution time.
Thanks again.

**Comment by mtmorgan** [comment]
The 10 min 15s is because the job was killed after 10 m, not because it took exactly 10m 15s to complete. The build machines can be under heavy load; apparently that is what happened on the Mac builder. You could rebuild or ignore the timeout, but if you're at the boundary (the [guidelines](http://bioconductor.org/developers/package-guidelines/#correctness) actually specify 5 minutes to build & check) then you'll likely run into these sorts of timeouts on a regular basis in the nightly builder, and it's better to further modify your vignette.

**Comment by mtmorgan** [comment]
So that we know where we stand, I see that you have made some good progress on getting your package into shape, but I assume that you are still working on your code since some of the initial problems remain, e.g.,

- inputs are not checked, e.g., Classification.R:93 it is unlikely that the function works on any 'data.frame'.
- there are still many uses of `1:...` (rather than `seq_len()`)

**Comment by BioinfoMonzino** [comment]
Dear @mtmorgan ,
thanks for your suggestions.
As you correctly supposed, we were working on our code to increase the control tests on data used as input for all the functions of our package. So, we have added further checks for the type of object and their contents. The user is advised with “warnings” and “stops” if the required conditions are not satisfied.
Furthermore, we modified the 1: … items with the use of seq_len().
A correction of some typos and a minor editing to the text of the vignette have also been done.
Regards.

**Comment by mtmorgan** [comment]
Again to be clear, I'll wait until you post a comment indicating that your package is ready before continuing the review process.

**Comment by BioinfoMonzino** [comment]
Dear @mtmorgan ,
this is a misunderstanding. Actually, we added a post after our last bump. We briefly summarized the modifications applied to our code following your suggestions. However, after the new submission, an error occurred due to a small bug in one of the change we did, but that was immediately fixed. We didn’t understand that we should have written another post.
By the way, the required corrections have been performed and the package is ready for the revision process.
Regards
